database-commons/src/main/java/co/cask/util/DBUtils.java

```diff
@@ -414,7 +419,17 @@ private static void unregisterOracleMBean(ClassLoader classLoader) {
   }
 
   public static String createConnectionString(ConnectionConfig config, String jdbcDriverName) {
-    return createConnectionString(config.host, config.port, config.database, jdbcDriverName);
+    Map<String, String> dbSpecificArguments = config.getDBSpecificArguments();
+    if (dbSpecificArguments.containsKey(ConnectionConfig.CONNECTION_STRING)) {
```

Comment:
Using special map keys and values is not a good idea. It also means the connection string will be used as a connection property which is not right.

This createConnectionString method does not belong in a central util class. Each individual config should be responsible for creating the connection, as they are the ones who have knowledge about the their jdbc driver expects.